Author: Justus Winter <4winter@informatik.uni-hamburg.de>
Date:   Mon Jul 8 15:56:46 2013 +0200
Description: killall5.c: Use sysconf(_SC_SYMLOOP_MAX) instead of MAXSYMLINKS.
 If sysconf returns an error, fall back to MAXSYMLINKS on platforms that
 define it.  Fixes build on Hurd.
Status:  applied upstream

--- a/src/killall5.c
+++ b/src/killall5.c
@@ -376,13 +376,26 @@ out:
 }
 
 /*
+ *     Get the maximal number of symlinks to follow.
+ */
+static int maxsymlinks(void)
+{
+	int v = sysconf(_SC_SYMLOOP_MAX);
+#ifdef MAXSYMLINKS
+	if (v == -1)
+		return MAXSYMLINKS;
+#endif
+	return v;
+}
+
+/*
  *     Check path is located on a network based partition.
  */
 int check4nfs(const char * path, char * real)
 {
 	char buf[PATH_MAX+1];
 	const char *curr;
-	int deep = MAXSYMLINKS;
+	int deep = maxsymlinks();
 
 	if (!nlist) return 0;
 
